tools/walletextension/wallet_extension.go

```diff
@@ -129,37 +132,43 @@ func (we *WalletExtension) handleHTTPEthJSON(resp http.ResponseWriter, req *http
 		return
 	}
 
-	// This is just a temporary unmarshalling. We need to unmarshall once to check if we got an error response, then
```

Comment:
In the new implementation, we encrypt the balance specifically, rather than the entire JSON object. This means we can improve on the old approach, where we had to tentatively unmarshal the JSON, and abort if the entire object was encrypted.